open_bus_stride_api/routers/route_timetable.py

```diff
@@ -0,0 +1,74 @@
+import typing
+import datetime
+
+import pydantic
+import sqlalchemy
+from fastapi import APIRouter
+
+from open_bus_stride_db.model.gtfs_stop import GtfsStop
+from open_bus_stride_db import model
+
+from . import common
+
+
+router = APIRouter()
+
+
+class RouteTimetablePydanticModel(pydantic.BaseModel):
+    id: int
+    name: str
+    city: str
+    lon: int
+    lat: int
+    planned_arrival_time: datetime.datetime = None
+    gtfs_line_ref: str = None
+    gtfs_line_start_time: datetime.datetime = None
+    gtfs_ride_id: str = None
+
+LIST_MAX_LIMIT = 100
+WHAT_PLURAL = """the stops timetable of a given bus.
+Currently, only planned time (gtfs) is returned for every stop"""
+TAG = 'user cases'
+PYDANTIC_MODEL = RouteTimetablePydanticModel
+
+def _post_session_query_hook(session_query: sqlalchemy.orm.Query):
+    return (
+        session_query
+        .select_from(model.GtfsStop)
+        .add_entity(model.GtfsRideStop)
+        .add_entity(model.GtfsRide)
+        .add_entity(model.GtfsRoute)
+        .join(model.GtfsRideStop, model.GtfsRideStop.gtfs_stop_id == model.GtfsStop.id)
+        .join(model.GtfsRide, model.GtfsRide.id == model.GtfsRideStop.gtfs_ride_id)
+        .join(model.GtfsRoute, model.GtfsRoute.id == model.GtfsRide.gtfs_route_id)
+        .order_by((sqlalchemy.asc)(model.GtfsRideStop.arrival_time))
+    )
+
+def _convert_to_dict(obj):
+    gtfs_stop, gtfs_ride_stop, gtfs_ride, gtfs_route = obj
+    return RouteTimetablePydanticModel(id=gtfs_stop.id, name=gtfs_stop.name, city=gtfs_stop.city, lat=gtfs_stop.lat,
+                                       lon=gtfs_stop.lon, planned_arrival_time=gtfs_ride_stop.arrival_time,
+                                       gtfs_line_ref=gtfs_route.line_ref, gtfs_line_start_time=gtfs_ride.start_time,
+                                       gtfs_ride_id=gtfs_ride.id).__dict__
+
+@common.router_list(router, TAG, PYDANTIC_MODEL, WHAT_PLURAL)
+def list_(limit: int = common.param_limit(LIST_MAX_LIMIT),
+          offset: int = common.param_offset(),
+          get_count: bool = common.param_get_count(),
+          planned_start_time_date_from: datetime.datetime = common.doc_param('planned_start_time', 'datetime_from', description='Set to get the timetable of a specific ride'),
+          planned_start_time_date_to: datetime.datetime = common.doc_param('planned_start_time', 'datetime_to', description='Set to get the timetable of a specific ride'),
```

Comment:
please add an index for this field to support more efficient queries

Reply:
Will do. How do you generate the new migration files? I see that you have a macro file (script.py.mako), but do you have a script that calcs the upgrade & downgrade or should I do it manually?

Reply:
I'm getting this error from alembic and the migration file is created empty - 
`/Users/shayadler/Hasadna/OpenBus/open-bus-stride-db/venv/lib/python3.9/site-packages/alembic/ddl/postgresql.py:221: UserWarning: autogenerate skipping functional index idx_gtfs_ride_start_time; not supported by SQLAlchemy reflection
  util.warn(
/Users/shayadler/Hasadna/OpenBus/open-bus-stride-db/venv/lib/python3.9/site-packages/alembic/ddl/postgresql.py:221: UserWarning: autogenerate skipping functional index idx_gtfs_ride_stop_arrival_time; not supported by SQLAlchemy reflection
  util.warn(
/Users/shayadler/Hasadna/OpenBus/open-bus-stride-db/venv/lib/python3.9/site-packages/alembic/ddl/postgresql.py:221: UserWarning: autogenerate skipping functional index idx_siri_ride_scheduled_start_date; not supported by SQLAlchemy reflection
  util.warn(
`

Note that the error is written both on my new indexes and also on an existing one (idx_siri_ride_scheduled_start_date), even though I saw in the docs that --autogenerate should identify new indexes. Happened to you? Anyway I'll create the file manually..

Reply:
I opened an issue, would be helpful if you could write full reproduction steps

https://github.com/hasadna/open-bus-stride-db/issues/14

I'll check it